### PR TITLE
New tests, for basic messages parsing, and sending messages to logging

### PR DESCRIPTION
**Pull request by weirdGuy**
Created new tests for parsing messages from chat stream, processing them
Also created tests for gdoc and gitter sending messages

### CODE REVIEW

File: test/messages-test.js [hunk 1]
```diff
@@ -1,40 +1,48 @@
...
+    it('Formatting message', function () {
+        save = sinon.stub(messages, 'getRoom').callsFake(function (roomId, callback) {
+            sinon.assert.calledOnce(save)
+            callback("testname")
         })
 
-        messages.sendMessage(message, "dest", function (inf) {})
+        messages.formatMessage({"text": "+todo @test this", "room": "testingName", "user": {"name": "test", "login": "test"}}, function (data) {
+        })
+        save.reset()
```
Comment (zelima) [inline test/messages-test.js:24]: I'm not really getting were/how the assertion happen here, with `save.reset()`? Would the tests fail if you do it before `messsages.formatMessage(...)`￼

File: test/messages-test.js [hunk 2]
```diff
@@ -1,40 +1,48 @@
...
 
-        sinon.assert.calledOnce(messages.sendMessage)
+    it('GDoc Logging', function () {
+        save = sinon.stub(messages, 'formatMessage').callsFake(function (message, callback) {
+            sinon.assert.calledOnce(save)
+        })
+        messages.sendMessage({"text": "+todo @test this", "user":{"name": "test", "login": "test"}}, "gdoc", function (data) {
 
+        })
+        save.reset()
```
Comment (zelima) [inline test/messages-test.js:34]: same here

File: test/messages-test.js [hunk 3]
```diff
@@ -1,40 +1,48 @@
...
-            return "0"
+    let save
+
+    beforeEach(function () {
+        save = null
+    })
+
+    it('Formatting message', function () {
+        save = sinon.stub(messages, 'getRoom').callsFake(function (roomId, callback) {
+            sinon.assert.calledOnce(save)
```
Comment (zelima) [inline test/messages-test.js:18]: Can we also test that this function was call with correct/expected parameters?
```
{
  "action": "expected",
  "timestamp": "expected",
  "poster": "expected",
  "assignees": "expected",
  "message": "expected",
  "room": "expected"
}
```
Eg: 
- test against expected output
- test when +todo is missing it should not be called at all
- test when it is in the middle

File: test/messages-test.js (hunk 1 above)
Comment (weirdGuy) [reply test/messages-test.js:24]: assertion is called inside stubbed function

Comment (zelima) [reply test/messages-test.js:24]: Can we do this outside the function?

**Review by zelima — CHANGES_REQUESTED** [review]
Also, following the README `npm test` throws an error `Error: Cannot find module 'coffee-script/register'`

Seems we have to install `coffee-script/register` and  `chai` additionally.

Can we fix this, or at least update instructions in README

**Comment by weirdGuy** [comment]
@zelima added commit with readme edition

**Comment by zelima** [comment]
Closing this PR. Tests needs the careful refactor as we are still not actually testing anything. Eg
There is no difference if I remove `messages.sendMessage({....})` part from this snippet tests will still be passing
```
  it('GDoc Logging', function () {
        save = sinon.stub(messages, 'formatMessage').callsFake(function (message, callback) {
            sinon.assert.calledOnce(save)
        })
        messages.sendMessage({"text": "+todo @test this", "user":{"name": "test", "login": "test"}}, "gdoc", function (data) {

        })
        save.reset()
    })
```

We need proper mocking of things.

**Pull request closed without merging**